src/Raven.Server/Documents/Replication/DocumentReplicationLoader.cs

```diff
@@ -31,36 +36,53 @@ protected override void LoadConfigurations()
-                //also, if there are destinations with multiple non-unique urls, add relevant alert
-
-                //something like : 
-                //				_database.AddAlert(new Alert
-                //				{
-                //					Title = "Multiple non-unique destinations are configured. Using the first one.",
-                //					CreatedAt = DateTime.UtcNow
-                //				});
-
-                if (configuration.Destinations == null) //precaution, should not happen
+                try
                 {
-                    _log.Warn("Invalid configuration document, Destinations property must not be null. Replication will not be active");
-                    return;
-                }
-
-                foreach (var destinationConfig in configuration.Destinations.Take(MaxSupportedReplicationDestinations))
-                {					
-                    var replicationExecuter = new DocumentReplicationExecuter(_database, destinationConfig);
-                    Replications.Add(replicationExecuter);
-                    replicationExecuter.Start();
+                    _replicationDocument = JsonDeserialization.ReplicationDocument(configurationDocument.Data);
+                    //the destinations here are the ones that are outbound..
+                    if (_replicationDocument.Destinations == null) //precaution, should not happen
+                    {
+                        _log.Warn(
+                            "Invalid configuration document, Destinations property must not be null. Replication will not be active");
+                    }
                 }
+                catch (Exception e)
+                {
+                    _log.Error("failed to deserialize replication configuration document. This is something that is not supposed to happen. Reason:" + e);
+                }	           
             }
         }
+
+        //inbound replication source will get it's ReplicationExecuter as well as
+        //the outgoing ones
+        public DocumentReplicationExecuter RegisterNewConnectionFrom(
+            Guid srcDbId, 
+            string srcUrl, 
+            string srcDbName, 
+            long lastSentEtag, 
+            out bool shouldConnectBack)
+        {
+            shouldConnectBack = false;
+
+            //prevent initiating two connections to the same db
+            if (!_activeConnections.TryAdd(srcDbId))
+                return null;
+
+            var destination = _replicationDocument.Destinations
+                .FirstOrDefault(x => x.Url.Equals(srcUrl, StringComparison.OrdinalIgnoreCase) &&
+                                     x.Database.Equals(srcDbName, StringComparison.OrdinalIgnoreCase));
+
+            if (destination != null)
+                shouldConnectBack = true;
+
+            var documentReplicationExecuter = new DocumentReplicationExecuter(_database, srcDbId, destination, lastSentEtag);
```

Comment:
In general, this should be happening in the course of processing the incoming request, all you need is to force the replications to do refresh all inactive replications.